extensions/resteasy-reactive/rest-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/BasicServerJacksonMessageBodyWriter.java

```diff
@@ -17,26 +19,46 @@
 import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter;
 import org.jboss.resteasy.reactive.server.spi.ServerRequestContext;
 
+import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectWriter;
 
 public class BasicServerJacksonMessageBodyWriter extends ServerMessageBodyWriter.AllWriteableMessageBodyWriter {
 
     private final ObjectWriter defaultWriter;
+    private final Map<Type, ObjectWriter> genericWriters = new ConcurrentHashMap<>();
 
     @Inject
     public BasicServerJacksonMessageBodyWriter(ObjectMapper mapper) {
         this.defaultWriter = createDefaultWriter(mapper);
     }
 
+    private ObjectWriter getWriter(Type genericType) {
+        if (genericType == null) {
+            // no explicit generic type. Use the default writer.
+            return this.defaultWriter;
+        } else {
+            // compute and cache a specific writer for the given generic type
+            return genericWriters.computeIfAbsent(genericType, type -> {
```

Comment:
For CHM, it's usually recommended to do a get first when getting a hit is the most common pattern as I was told `computeIfAbsent` locks were a bit less optimized.

Reply:
- lambda removed
- get() before computeIfAbsent() added

Reply:
I am still not 100% happy with the solution, because I feel I am generating too many custom writers (basically one per different method return type), even though that's probably only necessary for methods that return generic collections etc. (basically all those methods where `method.getGenericReturnType() <> method.getReturnType()). For methods with non-generic return types, the default writer works correctly.
I will dig just a little bit deeper to see whether I can differentiate those cases and only generate a custom writer if it is _really_ necessary.

Reply:
I have checked the details of how and why Jackson works this way and performed various other tests. As commented in the code, Jackson needs that custom writer exactly in those cases, where the target type is a parameterized type.
So I have updated the code once more to compute a specialized writer only in those cases, where the method return type actually is a generic (parameterized) type. In all other cases, the default writer can be used.

Imho, the code is good to go.

BTW: I also compared my solution to what is being done in FullyFeaturedServerJacksonMessageBodyWriter (which computes a customer writer per method/type as well). One difference that I spotted is that FullyFeaturedServerJacksonMessageBodyWriter maps the type by name (perTypeWriter in line 40) instead of using the type itself as the map key. I don't see, why this would be beneficial (on the contrary: computing the type name every time should be more expensive that using hashCode() and equals() of the type itself, right?!), but please correct me if I'm wrong.